resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/builders/ParallelBuildChainTest.java

```diff
@@ -228,9 +222,10 @@ public void testWorkspaceBuild_ConflictingRule() throws Exception {
 		executeIncrementalWorkspaceBuild(() -> {
 			waitForCondition(() -> TimerBuilder.getStartedProjectBuilds().size() > 1,
 					millisToWaitForUnexpectedParallelBuild);
-			assertThat(
-					"all build jobs have started in time although infinitely running builds with conflicting rules exist",
-					TimerBuilder.getStartedProjectBuilds(), not(containsInAnyOrder(longRunningProjects)));
+			assertThat(longRunningProjects).as(
+					"all build jobs have started in time although infinitely running builds with conflicting rules exist")
+					.anySatisfy(longRunningProject -> assertThat(TimerBuilder.getStartedProjectBuilds())
+							.doesNotContain(longRunningProject));
```

Comment:
```suggestion
			assertThat(longRunningProjects)
            assertThat(TimerBuilder.getStartedProjectBuilds())
                            .withFailMessage("all build jobs have started in time although infinitely running builds with conflicting rules exist")
							.doesNotContainAnyOf(longRunningProjects);
```

Reply:
Thank you for your feedback, @Bananeweizen!
I also thought some time about the `as(message)` calls, in particular because some of the existing failure messages do not really make sense as a description for the assertion. Many of them would totally make sense when passed to `withFailureMessage(message)`, but then the failure report will _only_ contain that message and nothing about the expected/actual value. In many cases it can be helpful to have more detailed information about the actual and expected values. However, for pure boolean assertions it totally makes sense to simply use `withFailureMessage()`, which is why I will update the PR according.

Regarding your concrete proposal for one assertion: the proposed assertion is semantically different from the original one. The original one checks that "not all long running project builds have started". The proposed on checks that "none of the long running project builds have started". This assertion looked better with Hamcrest, where you can combine matchers to: `assertThat(TimerBuilder.getStartedProjectBuilds(), not(containsInAnyOrder(longRunningProjects)))`. Unfortunately, I did not find an according assertion in AssertJ, which is why I made this strange construct of assertions realizing "there is a long running project build that has not started". If you know a better way to express that, I'd be happy if you can let me know :-)